src/services/balancer/entities/pools.ts

```diff
@@ -0,0 +1,157 @@
+import { PoolType } from '@/services/balancer/subgraph/types';
+
+import {
+  Vault__factory,
+  WeightedPoolFactory__factory,
+  WeightedPool__factory
+} from '@balancer-labs/typechain';
+import { Contract } from '@ethersproject/contracts';
+import { TransactionResponse, TransactionReceipt, Web3Provider } from '@ethersproject/providers';
+import { configService } from '@/services/config/config.service';
+import BigNumber from 'bignumber.js';
+import { BigNumber as EPBigNumber } from '@ethersproject/bignumber';
+import { sendTransaction } from '@/lib/utils/balancer/web3';
+import { defaultAbiCoder } from '@ethersproject/abi';
+import { TokenWeight } from '@/composables/pools/usePoolCreation';
+import { toNormalizedWeights } from '@balancer-labs/balancer-js';
+import { scale } from '@/lib/utils';
+
+type Address = string;
+
+export interface CreatePoolReturn {
+  id: string;
+  address: Address;
+}
+
+const JOIN_KIND_INIT = 0;
+
+export interface JoinPoolRequest {
+  assets: Address[];
+  maxAmountsIn: string[];
+  userData: any;
+  fromInternalBalance: boolean;
+}
+
+export default class Pools {
+  create(type: PoolType) {
+    switch (type) {
+      case PoolType.Weighted:
+        return this.createWeightedPool.bind(this);
+      default:
+        throw new Error('Cannot create pool of type: ' + type);
+    }
+  }
+
+  public async createWeightedPool(
+    provider: Web3Provider,
+    name: string,
+    symbol: string,
+    swapFee: string,
+    tokens: TokenWeight[],
+    owner: Address
+  ): Promise<CreatePoolReturn> {
+    if (!owner.length) return Promise.reject('No pool owner specified');
+
+    const weightedPoolFactoryAddress =
+      configService.network.addresses.weightedPoolFactory;
+
+    // Tokens must be sorted in order of addresses
+    tokens = this.sortTokens(tokens);
+
+    const tokenAddresses: Address[] = tokens.map((token: TokenWeight) => {
+      return token.tokenAddress;
+    });
+
+    const tokenWeights = this.calculateTokenWeights(tokens);
+    const swapFeeScaled = scale(new BigNumber(swapFee), 16);
+
+    const params = [
+      name,
+      symbol,
+      tokenAddresses,
+      tokenWeights,
+      swapFeeScaled.toString(),
+      owner
+    ];
+
+    const tx: TransactionResponse = await sendTransaction(
+      provider,
+      weightedPoolFactoryAddress,
+      WeightedPoolFactory__factory.abi,
+      'create',
+      params
+    );
+
+    const receipt: any = await tx.wait();
+    const events = receipt.events.filter(e => e.event === 'PoolCreated');
+    const poolAddress = events[0].args[0];
+
+    const pool = new Contract(poolAddress, WeightedPool__factory.abi, provider);
+    const poolId = await pool.getPoolId();
+
+    const poolDetails: CreatePoolReturn = {
+      id: poolId,
+      address: poolAddress
+    };
+
+    return poolDetails;
+  }
+
+  public async join(
```

Comment:
Since this pool service may eventually contain all pool related logic I think we should call this `initJoin` so we are being specific about the type of join. We have other logic elsewhere about different types of joins that may move here if it makes sense.